mail/api.py

```diff
@@ -337,3 +332,28 @@ def mark_emails_as_sent(automatic_email, emails):
             user=user,
             automatic_email=automatic_email,
         )
+
+
+def get_mail_vars(emails):
+    """
+    Returns a generator of mail template variables for each email in emails
+
+    Args:
+        emails (iterable of str): A list of email addresses
+
+    Returns:
+        generator of dict:
+            A dictionary of template variables along with email so we can tell who is who
+    """
+    queryset = User.objects.filter(email__in=emails).values(
+        'email',
+        'profile__mail_id',
+        'profile__preferred_name',
+    )
+    return (
```

Comment:
Since you're returning a generator here, I would probably suggest using `iterator()` so that the entire result isn't loaded into memory at start of generator iteration. If only `OrderedDict` generators were a thing, that would be even nicer :smile: